grails-app/controllers/io/xh/toolbox/contacts/ContactsController.groovy

```diff
@@ -0,0 +1,40 @@
+package io.xh.toolbox.contacts
+
+import io.xh.hoist.config.AppConfig
+import io.xh.hoist.exception.NotAuthorizedException
+import io.xh.hoist.security.Access
+import io.xh.toolbox.BaseController
+
+import static io.xh.hoist.json.JSONSerializer.serializePretty
+
+
+@Access(['APP_READER'])
+class ContactsController extends BaseController {
+
+    def configService
+
+    def index() {
+        renderJSON(getContacts())
+    }
+
+    def update(String id) {
+        if (!user.isHoistAdmin) throw new NotAuthorizedException()
```

Comment:
@lbwexler do we support `@Access` annotations at the method level?  I assumed not - is this the best practice approach if we want to avoid creating an entirely different controller?

Reply:
yep, you got it. So I think this is what we want -- yes!